src/screens/Messages/List/ChatListItem.tsx

```diff
@@ -98,11 +104,42 @@ function ChatListItemReady({
   let lastMessage = _(msg`No messages yet`)
   let lastMessageSentAt: string | null = null
   if (ChatBskyConvoDefs.isMessageView(convo.lastMessage)) {
-    if (convo.lastMessage.sender?.did === currentAccount?.did) {
-      lastMessage = _(msg`You: ${convo.lastMessage.text}`)
-    } else {
-      lastMessage = convo.lastMessage.text
+    const isFromMe = convo.lastMessage.sender?.did === currentAccount?.did
+
+    if (convo.lastMessage.text) {
+      if (isFromMe) {
+        lastMessage = _(msg`You: ${convo.lastMessage.text}`)
+      } else {
+        lastMessage = convo.lastMessage.text
+      }
+    } else if (convo.lastMessage.embed) {
+      const defaultEmbeddedContentMessage = _(msg`(contains embedded content)`)
+
+      if (AppBskyEmbedRecord.isView(convo.lastMessage.embed)) {
+        const embed = convo.lastMessage.embed
+
+        if (AppBskyEmbedRecord.isViewRecord(embed.record)) {
+          const record = embed.record
+          const path = postUriToRelativePath(record.uri, {
+            handle: record.author.handle,
+          })
+          const href = path ? toBskyAppUrl(path) : undefined
+          const short = href ? toShortUrl(href) : defaultEmbeddedContentMessage
+          if (isFromMe) {
+            lastMessage = _(msg`You: ${short}`)
+          } else {
+            lastMessage = short
+          }
+        }
+      } else {
+        if (isFromMe) {
+          lastMessage = _(msg`You: ${defaultEmbeddedContentMessage}`)
+        } else {
```

Comment:
small nit, maybe we can just add the `isFromMe` check once at end of these other checks. i think we'll probably have cases here eventually with external links, photos, etc.